### PR TITLE
use utils/ptr package instead of utils/pointer

### PR DESCRIPTION
**Pull request by sivchari**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

I made kubelet's codebase use ptr pacakge instead of pointer package.
This PR is only cleanup and these changes don't affect existing performance, so I modified apis package.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -447,7 +447,7 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 	}, {
 		name: "invalid MemoryThrottlingFactor",
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.MemoryThrottlingFactor = utilpointer.Float64(1.1)
+			conf.MemoryThrottlingFactor = ptr.To(1.1)
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/validation/validation_test.go:450]: Should it be `ptr.To[float64](1.1)`?

Comment (sivchari) [reply pkg/kubelet/apis/config/validation/validation_test.go]: I fixed it. Thanks.

File: pkg/kubelet/apis/config/fuzzer/fuzzer.go [hunk 2]
```diff
@@ -118,7 +118,7 @@ func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
 				obj.Logging.Format = "text"
 			}
 			obj.EnableSystemLogHandler = true
-			obj.MemoryThrottlingFactor = utilpointer.Float64(rand.Float64())
+			obj.MemoryThrottlingFactor = ptr.To(rand.Float64())
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/fuzzer/fuzzer.go:121]:
```suggestion
			obj.MemoryThrottlingFactor = ptr.To[float64](rand.Float64())
```

File: pkg/kubelet/apis/config/fuzzer/fuzzer.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
 				obj.Logging.Format = "text"
 			}
 			obj.EnableSystemLogHandler = true
-			obj.MemoryThrottlingFactor = utilpointer.Float64(rand.Float64())
+			obj.MemoryThrottlingFactor = ptr.To[float64](rand.Float64())
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/fuzzer/fuzzer.go:121]:
```suggestion
			obj.MemoryThrottlingFactor = ptr.To(rand.Float64())
```

Comment (sivchari) [reply pkg/kubelet/apis/config/fuzzer/fuzzer.go]: fix.

File: pkg/kubelet/apis/config/v1beta1/defaults.go [hunk 4]
```diff
@@ -257,25 +257,25 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
...
-		obj.EnableDebugFlagsHandler = utilpointer.Bool(true)
+		obj.EnableDebugFlagsHandler = ptr.To(true)
 	}
 	if obj.SeccompDefault == nil {
-		obj.SeccompDefault = utilpointer.Bool(false)
+		obj.SeccompDefault = ptr.To(false)
 	}
 	if obj.MemoryThrottlingFactor == nil {
-		obj.MemoryThrottlingFactor = utilpointer.Float64(DefaultMemoryThrottlingFactor)
+		obj.MemoryThrottlingFactor = ptr.To(DefaultMemoryThrottlingFactor)
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/v1beta1/defaults.go:272]:
```suggestion
		obj.MemoryThrottlingFactor = ptr.To[float64](DefaultMemoryThrottlingFactor)
```

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go [hunk 5]
```diff
@@ -64,25 +64,25 @@ func TestSetDefaultsKubeletConfiguration(t *testing.T) {
...
-				EventRecordQPS:                            utilpointer.Int32(50),
+				EventRecordQPS:                            ptr.To[int32](50),
 				EventBurst:                                100,
-				EnableDebuggingHandlers:                   utilpointer.Bool(true),
-				HealthzPort:                               utilpointer.Int32(10248),
+				EnableDebuggingHandlers:                   ptr.To(true),
+				HealthzPort:                               ptr.To[int32](10248),
 				HealthzBindAddress:                        "127.0.0.1",
-				OOMScoreAdj:                               utilpointer.Int32(int32(qos.KubeletOOMScoreAdj)),
+				OOMScoreAdj:                               ptr.To(int32(qos.KubeletOOMScoreAdj)),
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/v1beta1/defaults_test.go:74]:
```suggestion
				OOMScoreAdj:                               ptr.To[int32](qos.KubeletOOMScoreAdj),
```

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go [hunk 6]
```diff
@@ -121,20 +121,20 @@ func TestSetDefaultsKubeletConfiguration(t *testing.T) {
...
-				EnableDebugFlagsHandler:       utilpointer.Bool(true),
-				SeccompDefault:                utilpointer.Bool(false),
-				MemoryThrottlingFactor:        utilpointer.Float64(DefaultMemoryThrottlingFactor),
-				RegisterNode:                  utilpointer.Bool(true),
-				LocalStorageCapacityIsolation: utilpointer.Bool(true),
+				EnableSystemLogHandler:        ptr.To(true),
+				EnableProfilingHandler:        ptr.To(true),
+				EnableDebugFlagsHandler:       ptr.To(true),
+				SeccompDefault:                ptr.To(false),
+				MemoryThrottlingFactor:        ptr.To(DefaultMemoryThrottlingFactor),
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/v1beta1/defaults_test.go:128]:
```suggestion
				MemoryThrottlingFactor:        ptr.To[float64](DefaultMemoryThrottlingFactor),
```

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go [hunk 7]
```diff
@@ -745,32 +745,32 @@ func TestSetDefaultsKubeletConfiguration(t *testing.T) {
...
-				EnableDebugFlagsHandler:       utilpointer.Bool(true),
-				SeccompDefault:                utilpointer.Bool(false),
-				MemoryThrottlingFactor:        utilpointer.Float64Ptr(DefaultMemoryThrottlingFactor),
-				RegisterNode:                  utilpointer.Bool(true),
-				LocalStorageCapacityIsolation: utilpointer.Bool(true),
+				EnableSystemLogHandler:        ptr.To(true),
+				EnableProfilingHandler:        ptr.To(true),
+				EnableDebugFlagsHandler:       ptr.To(true),
+				SeccompDefault:                ptr.To(false),
+				MemoryThrottlingFactor:        ptr.To(DefaultMemoryThrottlingFactor),
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/v1beta1/defaults_test.go:752]:
```suggestion
				MemoryThrottlingFactor:        ptr.To[float64](DefaultMemoryThrottlingFactor),
```

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go [hunk 8]
```diff
@@ -873,24 +873,24 @@ func TestSetDefaultsKubeletConfiguration(t *testing.T) {
...
-				EventRecordQPS:                            utilpointer.Int32Ptr(50),
+				EventRecordQPS:                            ptr.To[int32](50),
 				EventBurst:                                100,
-				EnableDebuggingHandlers:                   utilpointer.Bool(true),
-				HealthzPort:                               utilpointer.Int32Ptr(10248),
+				EnableDebuggingHandlers:                   ptr.To(true),
+				HealthzPort:                               ptr.To[int32](10248),
 				HealthzBindAddress:                        "127.0.0.1",
-				OOMScoreAdj:                               utilpointer.Int32Ptr(int32(qos.KubeletOOMScoreAdj)),
+				OOMScoreAdj:                               ptr.To(int32(qos.KubeletOOMScoreAdj)),
```
Comment (bart0sh) [inline pkg/kubelet/apis/config/v1beta1/defaults_test.go:883]:
```suggestion
				OOMScoreAdj:                               ptr.To[int32](qos.KubeletOOMScoreAdj),
```

File: pkg/kubelet/images/image_manager_test.go [hunk 9]
```diff
@@ -473,7 +473,7 @@ func TestMaxParallelImagePullsLimit(t *testing.T) {
 	maxParallelImagePulls := 5
 	var wg sync.WaitGroup
 
-	puller, fakeClock, fakeRuntime, container, _ := pullerTestEnv(t, *testCase, useSerializedEnv, utilpointer.Int32Ptr(int32(maxParallelImagePulls)))
+	puller, fakeClock, fakeRuntime, container, _ := pullerTestEnv(t, *testCase, useSerializedEnv, ptr.To(int32(maxParallelImagePulls)))
```
Comment (bart0sh) [inline pkg/kubelet/images/image_manager_test.go:476]:
```suggestion
	puller, fakeClock, fakeRuntime, container, _ := pullerTestEnv(t, *testCase, useSerializedEnv, ptr.To[int32](maxParallelImagePulls))
```

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go (hunk 8 above)
Comment (sivchari) [reply pkg/kubelet/apis/config/v1beta1/defaults_test.go:883]: Same above.

File: pkg/kubelet/images/image_manager_test.go (hunk 9 above)
Comment (sivchari) [reply pkg/kubelet/images/image_manager_test.go:476]: Same above reason.

File: pkg/kubelet/apis/config/v1beta1/defaults_test.go (hunk 5 above)
Comment (bart0sh) [reply pkg/kubelet/apis/config/v1beta1/defaults_test.go:74]: makes sense to me, thanks for pointing out

**Comment by bart0sh** [comment]
@sivchari Please, rebase. Otherwise lgtm.

**Comment by sivchari** [comment]
@bart0sh 
Thanks, I fixed it.

**Comment by bart0sh** [comment]
@sivchari Please, squash commits into one, thanks!

**Comment by sivchari** [comment]
@bart0sh 
I squased commits into one, thanks.

**Comment by bart0sh** [comment]
/lgtm

/assign @mrunalp @SergeyKanzhelev 
for approval

**Comment by SergeyKanzhelev** [comment]
> I made kubelet's codebase use ptr pacakge instead of pointer package.
> This PR is only cleanup and these changes don't affect existing performance, so I modified apis package.

pardon my ignorence. Is some of it being deprecated? Any links to help understand why this is needed will help

**Comment by aojea** [comment]
@SergeyKanzhelev utils/pointer is just left for backward compatibility, but you can see it calls `ptr.To` internally https://github.com/kubernetes/utils/pull/283/

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Review by hashim21223445 — APPROVED** [review]

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**